tokio-fs/src/lib.rs

```diff
@@ -1,16 +1,29 @@
-//! Asynchronous filesystem manipulation operations (and stdin, stdout, stderr).
+//! Asynchronous file and standard stream adaptation.
 //!
-//! This module contains basic methods and types for manipulating the contents
-//! of the local filesystem from within the context of the Tokio runtime.
+//! This module contains utility methods and adapter types for input/output to
+//! files or standard streams (`Stdin`, `Stdout`, `Stderr`), and
+//! filesystem manipulation, for use within (and only within) a Tokio runtime.
 //!
-//! Tasks running on the Tokio runtime are expected to be asynchronous, i.e.,
-//! they will not block the thread of execution. Filesystem operations do not
-//! satisfy this requirement. In order to perform filesystem operations
-//! asynchronously, this library uses the [`blocking`][blocking] annotation
-//! to signal to the runtime that a blocking operation is being performed. This
-//! allows the runtime to compensate.
+//! Tasks run by *worker* threads should not block, as this could delay
+//! servicing reactor events. Portable filesystem operations are blocking,
+//! however. This module offers adapters which use a [`blocking`] annotation
+//! to inform the runtime that a blocking operation is required. When
+//! necessary, this allows the runtime to convert the current thread from a
+//! *worker* to a *backup* thread, where blocking is acceptable.
 //!
-//! [blocking]: https://docs.rs/tokio-threadpool/0.1/tokio_threadpool/fn.blocking.html
+//! ## Usage
+//!
+//! Where possible, users should prefer the provided asynchronous-specific
```

Comment:
Ah, you were talking about reading and writing to `tokio_fs::File` in particular. Would this be better in the `File` documentation instead of the crate level?

Reply:
Here is a (*marketing brochure*) example of that, here:

https://github.com/hyperium/hyper/blob/master/examples/send_file.rs#L66

**Q:** Does it handle `WouldBlock` appropriately?
**A:** No.

Reply:
I've inclued it at crate level because I think it applies to most anything from *tokio-fs* that returns an `io::Error` (e.g.`WouldBlock`).

Reply:
Doesn't `tokio_io::io::read_to_end` handle that? It uses `try_nb!`, which converts `ErrorKind::WouldBlock` to `Poll::NotReady`.

Reply:
I'm not 100% certain so I'll stick to my "people will gloss over `WouldBlock`" as my justification, at least until, I can be certain or find better examples of the problem (besides my own code!).

Reply:
I was probably wrong in my initial Q/A, since `tokio_io::io::read_to_end` returns a `Future`, but the rustdoc changes make a clear distincation between that an anything still returining `io::Error`.

Reply:
Yes, one should not use `std::io::Read::read` and instead use `AsyncRead::poll_read`.